### ISSUE TITLE
Getting intermittent HTTP2 framing errors when aborting long running connections

### ISSUE THREAD

**Issue by mccare** [fastify/help#489]
## 💬  Getting intermittent HTTP2 framing errors when aborting long running connections

I'm trying to stream data via fastify on Google Cloud Run. To do this on Cloud Run I need to enable http2 (h2c). But I do get intermittent 503 errors. I've now tested that locally and curl will report `A problem was detected in the HTTP2 framing layer` (error Code 16) Any ideas what I'm doing wrong here?

```js
const dataSize = 50 * 1024 * 1024
const data = Buffer.from(new Uint8Array(dataSize))

module.exports = async function (fastify, opts) {
  fastify.get('/storage/video/:file', async function (request, reply) {
    reply.code(200).send(data)
  })
}
````

Server is started with http2: true

Sample code to reproduce my error is here: https://github.com/mccare/http2downloaderrors
Basically starting a server, and then having two running request (long) and always adding to that new requests that will be cancled after 2s. Then I get intermittent http2 frame errors. If I do the same thing with a plain node http2 server, that works.

## Your Environment

- *node version*: 14.17.2
- *fastify version*: 3.19.1
- *os*: Mac 11.4

**Comment by mcollina** [comment]
Have you verified that the a server written in Node.js without fastify does not have this problem?

**Comment by mccare** [comment]
Yes. I've included both versions in the git repository. (with fastify and with node only).

**Comment by climba03003** [comment]
The problem is your testcase is not identical between NodeJS `http2` and `fastify`.

You have specified `http2SessionTimeout` on `fastify` and `http2` do not.
If you do the same on `http2`, it will occur on `http2`.

**Comment by mccare** [comment]
Thanks a lot, that fixed this problem. As a side note: http2SessionTimeout of 0 on cloud run did not work, I've now set it to 10 minutes and my errors seem to be gone.
